### ISSUE TITLE
UnauthorizedHandler never called

### ISSUE THREAD

**Issue by mstimvol** [yubinTW/fastify-keycloak-adapter#31]
At first... Thank you very much for this great plugin! It works perfectly - except one thing... my `unauthorizedHandler` is never called. Here's my registration:

```
  fastify.register(require('fastify-keycloak-adapter'), {
    appOrigin: env('APP_URL'),
    keycloakSubdomain: env('KEYCLOAK_SUBDOMAIN'),
    clientId: env('KEYCLOAK_CLIENT_ID'),
    clientSecret: env('KEYCLOAK_CLIENT_SECRET'),
    callback: '/login',
    autoRefreshToken: true,
    scope: ['openid', 'profile', 'email'],
    unauthorizedHandler: (req, resp) => {
      console.log('hello world!')
      resp.status(401).send('Invalid request')
    },
    userPayloadMapper: (payload) => ({
      name: payload.name,
      email: payload.email,
      given_name: payload.given_name,
      family_name: payload.family_name,
      preferred_username: payload.preferred_username
    })
  })
```

In my fastify logs I see only the following:

```
  {"level":20,"time":1691607443896,"pid":10120,"hostname":"PC-09","msg":"Keycloak Options valid successfully. Keycloak options: <removed ;-)>"}
  {"level":30,"time":1691607444242,"pid":10120,"hostname":"PC-09","msg":"Keycloak registered successfully!"}
  {"level":30,"time":1691607444342,"pid":10120,"hostname":"PC-09","msg":"Server listening at http://[::1]:8888"}
  {"level":30,"time":1691607444343,"pid":10120,"hostname":"PC-09","msg":"Server listening at http://127.0.0.1:8888"}
  Webserver is listening on http://[::1]:8888
  {"level":20,"time":1691607465215,"pid":10120,"hostname":"PC-09","reqId":"req-1","msg":"Keycloak adapter: grant not found in session"}
```

And then I'm getting redirected to Keycloak. Any idea what I'm doing wrong here?

**Comment by yubinTW** [comment]
yes, I have fixed this issue in `2.1.0`
when user set `unauthorizedHandler`, he has to handle all of the unauthorized situations.

**Comment by mstimvol** [comment]
Thank you very much, @yubinTW! I updated my application and now it works.
